sphinx/ext/napoleon/docstring.py

```diff
@@ -879,6 +1047,16 @@ def __init__(self, docstring: Union[str, List[str]], config: SphinxConfig = None
         self._directive_sections = ['.. index::']
         super().__init__(docstring, config, app, what, name, obj, options)
 
+    def _get_location(self) -> str:
+        filepath = inspect.getfile(self._obj) if self._obj is not None else ""
+        name = self._name
+        line = ""
+
+        if filepath is None and name is None:
+            return None
+
+        return ":".join([filepath, "docstring of %s" % name, line])
```

Comment:
It seems `line` is always empty. Why do you add it here?

Reply:
because I was hoping to add that information somehow, but couldn't find out how to do that. Also, the format of the location of other warnings about docstrings (the nitpicky ones?) seems to be "{path}:docstring of {func}:{line}", so to copy that, the empty string is necessary.

Reply:
We usually suppress a line number on the location when hard to obtain it. For example, our logging module suppresses it automatically:
https://github.com/sphinx-doc/sphinx/blob/5b096c42ff3a907fa7bcce3fa620209da16d40bd/sphinx/util/logging.py#L487-L490

Please remove it from this list.